### PR TITLE
feat: allow to skip publishing duplicate package

### PR DESCRIPTION
**Pull request by rokklobster**
A solution to #775. The PR adds new flag, checks for this flag on related errors

### CODE REVIEW

**Comment by rokklobster** [comment]
@microsoft-github-policy-service agree

File: src/main.ts [hunk 1]
```diff
@@ -197,6 +197,7 @@ module.exports = function (argv: string[]): void {
 		.option('--pre-release', 'Mark this package as a pre-release')
 		.option('--allow-star-activation', 'Allow using * in activation events')
 		.option('--allow-missing-repository', 'Allow missing a repository URL in package.json')
+		.option('--skip-duplicate', 'Do not fail if package already exists on the marketplace')
```
Comment (joaomoreno) [inline src/main.ts:200]:
```suggestion
		.option('--skip-duplicate', 'Fail silently if version already exists on the marketplace')
```

File: src/publish.ts [hunk 2]
```diff
@@ -57,6 +57,8 @@ export interface IPublishOptions {
 	readonly preRelease?: boolean;
 	readonly allowStarActivation?: boolean;
 	readonly allowMissingRepository?: boolean;
+
+	readonly skipDuplicate?: boolean;
```
Comment (joaomoreno) [inline src/publish.ts:61]:
```suggestion
	readonly skipDuplicate?: boolean;
```

File: src/publish.ts [hunk 3]
```diff
@@ -170,6 +172,10 @@ async function _publish(packagePath: string, manifest: Manifest, options: IInter
 			const sameVersion = extension.versions.filter(v => v.version === manifest.version);
 
 			if (sameVersion.length > 0) {
+				if (options.skipDuplicate) {
+					log.done(`Found v${manifest.version} - skipping publish`);
```
Comment (joaomoreno) [inline src/publish.ts:176]:
```suggestion
					log.done(`Version ${manifest.version} is already published. Skipping publish.`);
```

File: src/publish.ts [hunk 4]
```diff
@@ -187,7 +193,12 @@ async function _publish(packagePath: string, manifest: Manifest, options: IInter
 				await api.updateExtension(undefined, packageStream, manifest.publisher, manifest.name);
 			} catch (err: any) {
 				if (err.statusCode === 409) {
-					throw new Error(`${description} already exists.`);
+					if (options.skipDuplicate) {
+						log.done(`Found v${manifest.version} - skipping publish`);
```
Comment (joaomoreno) [inline src/publish.ts:197]:
```suggestion
						log.done(`Version ${manifest.version} is already published. Skipping publish.`);
```

**Review by joaomoreno — DISMISSED** [review]

**Review by aeschli — APPROVED** [review]

**Review by joaomoreno — APPROVED** [review]

**Pull request merged**